pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -108,6 +116,24 @@ func (c *oauthClientsController) sync(ctx context.Context, controllerContext fac
 
 	statusHandler := status.NewStatusHandler(c.operatorClient)
 
+	infrastructureConfig, err := c.infrastructureClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("infrastructure config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	clusterVersionConfig, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		klog.Errorf("cluster version config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// Disable the oauth client update for external control plane topology (hypershift) if the ingress capability is disabled.
+	// HyperShift handles the updating of the oauth client for the console.
+	if util.IsExternalControlPlaneWithIngressDisabled(infrastructureConfig, clusterVersionConfig) {
```

Comment:
by doing this you are basically disabling the operator from doing its work by syncing the console oauthclients.

Reply:
By "syncing the console oauthclients" you mean adding `redirectURI`? From what I understand this part will be handled by HyperShift as it takes care of some of the `OAuthClient`s already. I suppose it's because [the authentication server is not exposed via a route](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/optional-ingress-hypershift.md#authentication-operator).

@csrwng:  What do you think? Should be disable the updates of console's `OAuthClient` on HyperShift or we can keep it for the time being?

Reply:
Also, in the absence of the ingress operator the required RBAC won't be created for the components (including the console operator): [Component routes chapter of the EP](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/optional-ingress-hypershift.md#component-routes).

Reply:
Makes sense to disable updates of the OAuthClient

Reply:
[The new API](https://github.com/openshift/api/pull/1887) has been implemented to support an alternative ingress.